hbase-shell/src/main/ruby/shell/commands/dry_run_balancer.rb

```diff
@@ -0,0 +1,43 @@
+#
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Shell
+  module Commands
+    class DryRunBalancer < Command
+      def help
+        return <<-EOF
+Trigger a dry run of the cluster balancer. This will execute the balancer logic,
+without actually moving any regions in the cluster. The output of this is useful
+for debugging balancer changes, or even adjustments to balancer weights. You are
+effectively able to preview balancer changes to the cluster before actually
+executing them.
+This will NOT run if regions in transition. This also will NOT
+run if the balancer is enabled (as to make sure this logic does not clash with
+the actual balancer).
+EOF
+      end
+
+      def command()
+        did_balancer_run = !!admin.dryRunBalancer()
+        formatter.row([did_balancer_run.to_s])
+        did_balancer_run
+      end
```

Comment:
Going off of what I suggested above with a single `Admin#balance()` method which can supply arguments, I think it'd be great to just extend the existing `balancer` command and let the user pass in a dry-run flag.

Reply:
yes

Reply:
yes